src/gui/toolbarMenubar/model/ToolbarColorNames.cpp

```diff
@@ -61,14 +60,14 @@ void ToolbarColorNames::addColor(Color color, const std::string& name, bool pred
         this->predefinedColorNames[color] = name;
     } else {
         char colorHex[16];
-        sprintf(colorHex, "%06x", uint32_t{color});
+        snprintf(colorHex, sizeof(colorHex), "%06" PRIx32, uint32_t{color});
         g_key_file_set_string(this->config, "custom", colorHex, name.c_str());
     }
 }
 
 auto ToolbarColorNames::getColorName(Color color) -> std::string {
     char colorHex[16];
```

Comment:
If you are touching this, you could directly replace old plain c arrays with c++11 std::arrays.
Another idea would be to completely get rid of c-format `sprintf` functions with std::format, but this is still not available on most compilers. Or use `std::stringstream`. But usecases of `std::stringstream` will be replaced mostly by std::format either. So for now this change is good.